joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -216,11 +216,18 @@ class JointStatePublisher():
                     parent = param['parent']
                     factor = param.get('factor', 1)
                     offset = param.get('offset', 0)
+                    # Handle recursive mimic chain
+                    recursive_mimic_chain_joints = [name]
                     while parent in self.dependent_joints:
+                        if parent in recursive_mimic_chain_joints:
+                            error_message = "Found an infinite recursive mimic chain"
+                            rospy.logerr("%s: [%s, %s]", error_message, ', '.join(recursive_mimic_chain_joints), parent)
+                            sys.exit(-1)
+                        recursive_mimic_chain_joints.append(parent)
                         param = self.dependent_joints[parent]
                         parent = param['parent']
                         factor *= param.get('factor', 1)
-                        # the offset is relative only to the first parent
+                        offset = factor*offset + param.get('offset', 0)
```

Comment:
If the recursive formula is correct

    J_k = f_k * J_k-1 + o_k
        = f_k * f_k-1 * ... * f_2 * f_1 * J_0 +
          + f_k * (f_k-1 * (... * (f_2 * o_1 + o_2) + ...) + o_k-1) + o_k

It should be:
- `factor = factor_new * factor_old`
- `offset = factor_new * offset_old + offset_new`